src/GraphQL/Types/TypeInformation.cs

```diff
@@ -0,0 +1,259 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace GraphQL.Types
+{
+    /// <summary>
+    /// Parses a <see cref="System.Type"/> along with nullability information into its constituent parts
+    /// in preparation for creating a graph type that represents such a type.
+    /// </summary>
+    public class TypeInformation
+    {
+        /// <summary>
+        /// The member being inspected.
+        /// </summary>
+        public MemberInfo MemberInfo { get; }
+
+        /// <summary>
+        /// Indicates that this is an input type (an argument); false for output types.
+        /// </summary>
+        public bool IsInputType { get; }
+
+        /// <summary>
+        /// The underlying CLR type represented. This might be the underlying type of a <see cref="Nullable{T}"/>
+        /// or the underlying type of a <see cref="IEnumerable{T}"/>.
+        /// </summary>
+        public Type Type { get; set; }
```

Comment:
It is assumed that you can install an arbitrary type here, even nulable and then everything will be calculated?

Reply:
No this property should only be set to the underlying type.

Generally, the constructor disects the original type into parts:
- Type
- IsList
- IsNullable
- ListIsNullable

This makes it easy to perform the following by attributes:
- Make the type nullable or not
- Make the list type nullable or not
- Change the underlying data type without changing the nullability or list flags

Reply:
So maybe add some checks in `set` ? What if I set `typeof(List<int?>)` ?

Reply:
Well as of now, it is useful in that it is able to produce lists like `int[][]`, because the value of this type is passed to `GetGraphTypeFromType`.  I think we should consider this in another PR.